src/qibolab/designs/basic.py

```diff
@@ -1,8 +1,61 @@
 from dataclasses import dataclass, field
+from typing import Any, List, Optional
 
 from qibolab.instruments.abstract import AbstractInstrument
 
 
+@dataclass
+class Channel:
+    """Representation of physical wire connection (channel).
+
+    Name is used as a unique identifier for channels.
+    Channel objects are instantiated by :class:`qibolab.platforms.platform.Platform`,
+    but their attributes are modified and used by instrument designs.
+
+    Args:
+        name (str): Name of the channel as given in the platform runcard.
+
+    Attributes:
+        ports (list): List of tuples (controller (`str`), port (`int`))
+            specifying the QM (I, Q) ports that the channel is connected.
+        qubits (list): List of tuples (:class:`qibolab.platforms.utils.Qubit`, str)
+            for the qubit connected to this channel and the role of the channel.
+        Optional arguments holding local oscillators and related parameters.
+        These are relevant only for mixer-based insturment designs.
+    """
+
+    name: str
+
+    qubit: Optional["Qubit"] = field(default=None, init=False, repr=False)
+    ports: List[tuple] = field(default_factory=list, init=False)
+    local_oscillator: Any = field(default=None, init=False)
+    lo_frequency: float = field(default=0, init=False)
+    lo_power: float = field(default=0, init=False)
+    _offset: Optional[float] = field(default=None, init=False)
+    _filter: Optional[dict] = field(default=None, init=False)
+
+    @property
+    def offset(self):
+        if self._offset is None:
+            # operate qubits at their sweetspot unless otherwise stated
+            self._offset = self.qubit.sweetspot
```

Comment:
Even this one I missed before: is there a reason to set it instead of just return?
```suggestion
            return self.qubit.sweetspot
```